guides/v2.2/coding-standards/technical-guidelines.md

```diff
@@ -532,7 +532,75 @@ class View extends Template
+
+6.4.3.4. Each service interface should declare a single public method. Interface name should reflect task/action to be performed. Example `Magento\InventoryApi\Api\StockSourceLinksDeleteInterface::execute(array $links)`. The only exception is Repository API which may be added for convenience and must be limited to singular CRUD operations and `getList($searchCriteria)`.
+
+6.4.4. Service Method Signature
+
+6.4.4.1. Service contracts should not apply presentation layer formatting to the returned data.
+
+6.4.4.2. Service data interfaces represent domain entities and can reference related data interfaces.
+
+6.4.4.3. Service data interfaces must be implemented as DTOs with a set of symmetric getters and setters.
+
+6.4.4.3. Strict typing is enforced for Service and Data interfaces located under `MyCompany/MyModuleApi/Api`. Only the following types are allowed:
+
+* Scalar types: `string` (including Date and DateTime); `int`; `float`; `boolean`
+
+* Data interfaces
+
+* One-dimensional indexed arrays of scalars or data interfaces: for example `string[]`, `\MyCompany\MyModuleApi\Api\Data\SomeInterface[]`. Hash maps (associative arrays) are not supported
+
+* Optional scalars or data interfaces: `string|null`. It is prohibited to use just `null`
+
+6.4.4.4. Service contracts should support batch data whenever possible. For example, entity persisting interface should accept an array of entities to persist instead of a single entity. Customizations via plugins must be adjusted respectively. Batch retrieval operations must accept `SearchCriteriaInterface` and return `SearchResultInterface` to support pagination.
+
+6.4.4.5. Command operations (the ones which modify the state) must support asynchronous execution. Void must be used as return type, the status of the operation may be checked by polling of the respective query operation. Create operations must support UUID generated on the client side.
+
+6.4.4.6. Command operations (the ones which modify the state) must support asynchronous execution. Void must be used as return type, the status of the operation may be checked by polling of the respective query operation. Create operations must support UUID generated on the client side.
+
+6.4.4.7. Data objects returned by service contracts should be fully loaded to ensure consistency.
+
+6.4.5. Service Implementation
+
+6.4.5.1. Service implementations and plugins must not rely on storage-specific integrity features, such as foreign key constraints.
+
+6.4.5.2. Replace strategy should be used to persist main entity fields/attributes, child entities and relation links. 
+
+6.4.5.3. During update operations, Web APIs with `PATCH` HTTP method and all action controllers that accept entities should pre-load them first, merge request data on top of that and provide full data to service.
```

Comment:
Should this logic be part of the service?

Reply:
it should NOT be, the same as deciding how to guarantee atomicity of these changes and prevent data inconsistency (i.e. wrapping all the process into the transaction). 

This is the responsibility of the Infrastructure layer, but not the Domain one

Reply:
I think yes. This item was migrated from previous discussions, @buskamuza @maghamed do you know what is the history behind this item?